neuralmonkey/learning_utils.py

```diff
@@ -280,8 +280,18 @@ def run_on_dataset(tf_manager: TensorFlowManager,
     if postprocess is not None:
         for series_name, postprocessor in postprocess:
             postprocessed = postprocessor(dataset, result_data)
+            if not hasattr(postprocessed, '__len__'):
+                postprocessed = list(postprocessed)
+
             result_data[series_name] = postprocessed
 
+    # check output series lengths
+    for series_id, data in result_data.items():
+        if len(data) != len(dataset):
+            warn("Output '{}' for dataset '{}' has length {}, but "
```

Comment:
Tadyten `warn` je takovej nebezpečnej. Pokud budu mít nastavenej strict mód, tak mi to tady na tom padne ještě před výpisem a přijdu o důlkežitý informace, který by mi pomohly zjistit, co je špatně. Nevim, jestli je potřeba s tim něco dělat v rámci tohoto pull requestu.. Možná bych tam dal log a TODO s otázkou, co s tim. Množí se místa, na kterých bychom chtěli mít warn, ale nedáváme ho tam, protože nechceme, aby to se strictem padalo..